snuba/migrations/migrations.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from datetime import datetime
+from enum import Enum
+from importlib import import_module
+from typing import Optional, Sequence, Tuple
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster, CLUSTERS
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations.operations import Operation
+
+
+logger = logging.getLogger("snuba.migrations")
+
+LOCAL_TABLE_NAME = "migrations_local"
+DIST_TABLE_NAME = "migrations_dist"
+
+# Currently only local migrations are supported
+TABLE_NAME = LOCAL_TABLE_NAME
+
+
+class App(Enum):
+    SYSTEM = "system"
+    SNUBA = "snuba"
+
+
+class Status(Enum):
+    NOT_STARTED = "not_started"
+    IN_PROGRESS = "in_progress"
+    COMPLETED = "completed"
+
+
+class Migration:
+    is_dangerous: bool
+    dependencies: Optional[Sequence[Tuple[str, str]]]
+
+    def forwards_local(self) -> Sequence[Operation]:
+        raise NotImplementedError
+
+    def backwards_local(self) -> Sequence[Operation]:
+        raise NotImplementedError
+
+    def forwards_dist(self) -> Sequence[Operation]:
+        raise NotImplementedError
+
+    def backwards_dist(self) -> Sequence[Operation]:
+        raise NotImplementedError
+
+
+class Manager:
```

Comment:
Personal opinion, I am not a big fan of the `Manager` classes since that name does not convey a lot about the responsibility of the class which inn this case is pretty clear: "it runs migrations". So what about a name like Runner or just avoid the class altogether, it does not have any state after all, and just have run_migration as a top level function in this module ?

Reply:
I agree it's a bad name, I will rename it to Runner. Although a class isn't required currently I added it since it will be responsible for keeping the state of all of the migrations soon.